schematic/store/synapse.py

```diff
@@ -1467,24 +1578,89 @@ def upload_manifest_file(
                 + file_extension
             )
 
-        manifestSynapseFile = File(
-            metadataManifestPath,
-            description="Manifest for dataset " + datasetId,
-            parent=datasetId,
-            name=file_name_new,
+        try:
+            # Rename the file to file_name_new then revert
+            # This is to maintain the original file name in-case other code is
+            # expecting that the file exists with the original name
+            original_file_path = metadataManifestPath
+            new_file_path = os.path.join(
+                os.path.dirname(metadataManifestPath), file_name_new
+            )
+            os.rename(original_file_path, new_file_path)
+
+            manifest_synapse_file = self._store_file_for_manifest_upload(
+                new_file_path=new_file_path,
+                dataset_id=datasetId,
+                existing_file_name=file_name_full,
+                file_name_new=file_name_new,
+                restrict_manifest=restrict_manifest,
+            )
+            manifest_synapse_file_id = manifest_synapse_file.id
+
+        finally:
+            # Revert the file name back to the original
+            os.rename(new_file_path, original_file_path)
+
+            if manifest_synapse_file:
+                manifest_synapse_file.path = original_file_path
+
+        return manifest_synapse_file_id
```

Comment:
Doing this prevents the need to call `changeFileMetaData` as the file will be uploaded with the expected name. That function was/is super slow so this is a big gain.